actionpack/lib/action_dispatch/journey/formatter.rb

```diff
@@ -54,11 +55,12 @@ def clear
         def extract_parameterized_parts(route, options, recall, parameterize = nil)
           parameterized_parts = recall.merge(options)
 
-          keys_to_keep = route.parts.reverse.drop_while { |part|
+          keys_to_keep = route.parts.reverse_each.drop_while { |part|
             !options.key?(part) || (options[part] || recall[part]).nil?
           } | route.required_parts
 
-          (parameterized_parts.keys - keys_to_keep).each do |bad_key|
+          parameterized_parts.each do |bad_key, _|
+            next if keys_to_keep.include?(bad_key)
             parameterized_parts.delete(bad_key)
```

Comment:
Pretty sure the logic would be the same, I think you could do that `delete_if` is slightly faster...

```
Calculating -------------------------------------
        each; delete    35.166k i/100ms
           delete_if    36.416k i/100ms
-------------------------------------------------
        each; delete    478.026k (± 8.5%) i/s -      2.391M
           delete_if    485.123k (± 7.9%) i/s -      2.440M
```